### PR TITLE
Cleanup useViewportColumns hook

### PR DESCRIPTION
**Pull request by amanmahajan7**

### CODE REVIEW

File: src/hooks/useViewportColumns.ts [hunk 1]
```diff
@@ -24,26 +20,116 @@ export function useViewportColumns<R, SR>({
...
+    }
+
+    return {
+      columns: calculatedColumns,
+      lastFrozenColumnIndex,
+      totalFrozenColumnWidth,
+      totalColumnWidth: totalWidth,
+      groupBy
+    };
+  }, [columnWidths, defaultFormatter, defaultResizable, defaultSortable, minColumnWidth, rawColumns, rawGroupBy, viewportWidth]);
```
Comment (amanmahajan7) [inline src/hooks/useViewportColumns.ts:132]: No need to run the hook if `rowGrouper` is changed.

File: src/utils/columnUtils.ts [hunk 2]
```diff
@@ -1,132 +1,6 @@
...
-    columns: calculatedColumns,
-    lastFrozenColumnIndex,
-    totalFrozenColumnWidth,
-    totalColumnWidth: totalWidth,
-    groupBy
-  };
-}
-
-function getSpecifiedWidth<R, SR>(
+export function getSpecifiedWidth<R, SR>(
```
Comment (nstepien) [inline src/utils/columnUtils.ts:3]: I think we can move `getSpecifiedWidth` and `clampColumnWidth` into `useViewportColumns.ts`, they're not meant to be used for anything, so no need to "share" them.

File: src/hooks/useViewportColumns.ts [hunk 3]
```diff
@@ -102,3 +187,35 @@ export function useViewportColumns<R, SR>({
 
   return { columns, viewportColumns, totalColumnWidth, lastFrozenColumnIndex, totalFrozenColumnWidth, groupBy };
 }
+
+export function getSpecifiedWidth<R, SR>(
```
Comment (nstepien) [inline src/hooks/useViewportColumns.ts:191]:
```suggestion
function getSpecifiedWidth<R, SR>(
```

File: src/hooks/useViewportColumns.ts [hunk 4]
```diff
@@ -102,3 +187,35 @@ export function useViewportColumns<R, SR>({
...
+  if (typeof width === 'number') {
+    return width;
+  }
+  if (typeof width === 'string' && /^\d+%$/.test(width)) {
+    return Math.floor(viewportWidth * parseInt(width, 10) / 100);
+  }
+  return undefined;
+}
+
+export function clampColumnWidth<R, SR>(
```
Comment (nstepien) [inline src/hooks/useViewportColumns.ts:209]:
```suggestion
function clampColumnWidth<R, SR>(
```

**Review by nstepien — CHANGES_REQUESTED** [review]

**Review by nstepien — APPROVED** [review]

**Pull request merged**